src/core/renderers/canvas/utils/CanvasMaskManager.js

```diff
@@ -14,6 +14,48 @@ export default class CanvasMaskManager
     constructor(renderer)
     {
         this.renderer = renderer;
+        this._maskStates = [];
+    }
+
+    /**
+     * @param {PIXI.CanvasRenderer} renderer - The canvas renderer.
+     * @returns {Canvas} a new canvas object with the same attributes as the one used in the rendering context.
+     */
+    _copyCanvas(renderer)
+    {
+        const canvas = document.createElement('canvas');
```

Comment:
These need to be pooled, creating one (or more) each frame is going to be a lot of GC.